.github/workflows/build.yml

```
@@ -1,39 +1,314 @@
+
+    - name: Install JS Dependencies
+      if: ${{ steps.cache-node-modules.outputs.cache-hit != 'true' }}
+      run: jlpm --prefer-offline --frozen-lockfile
+
+    - name: Build JS Library
+      run: jlpm build:lib
+
+    - name: Build JupyterLab Extension
+      run: jlpm build:labextension
+
+    - name: Build PyPI Distributions
+      run: python setup.py sdist bdist_wheel
+
+    - name: Validate Python Distributions
+      run: twine check dist/*
+
+    - name: Collect JS Distribution
+      run: |
+        set -eux
+        cp jupyterlab_pullrequests/labextension/*.tgz dist
+
+    - name: Hash Distributions
+      run: |
+        set -eux
+        cd dist
+        sha256sum * | tee SHA256SUMS
+
+    - name: Upload Distributions
+      uses: actions/upload-artifact@v2
+      with:
+        name: pull-requests dist ${{ github.run_number }}
+        path: ./dist
+
+  lint:
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        os: [ubuntu]
+        python-version: [3.9]
+        node-version: [14.x]
+        lab-version: [2]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install NodeJS
+      uses: actions/setup-node@v2
+      with:
+       node-version: ${{ matrix.node-version }}
+
+    - name: Cache JS
+      id: cache-node-modules
+      uses: actions/cache@v2
       with:
-       node-version: '14.x'
+        path: node_modules
+        key: |
+          ${{ env.CACHE_EPOCH }}-node-modules-${{ runner.os }}-${{ hashFiles('yarn.lock', '.yarnrc') }}
+
+    - name: Install Python Setup Dependencies
+      run: python -m pip install --upgrade pip wheel setuptools
+
+    - name: Get Python Cache
+      id: cache-pip
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Cache Python Dependencies
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.cache-pip.outputs.dir }}
+        key: |
+          ${{ env.CACHE_EPOCH }}-${{ runner.os }}-pip-lint-${{ matrix.python-version }}-${{ hashFiles('setup.py', 'setup.cfg') }}
+        restore-keys: |
+          ${{ env.CACHE_EPOCH }}-${{ runner.os }}-pip-lint-${{ matrix.python-version }}-
+          ${{ env.CACHE_EPOCH }}-${{ runner.os }}-pip-
+
+    - name: Install Setup Dependencies
+      run: python -m pip install 'jupyterlab==${{ matrix.lab-version }}.*'
+
+    - name: Install JS Dependencies
+      if: ${{ steps.cache-node-modules.outputs.cache-hit != 'true' }}
+      run: jlpm --prefer-offline --frozen-lockfile
+
+    - name: Build JS
+      run: jlpm build
+
+    - name: Install Python Lint/Test Dependencies
+      run: python -m pip install -e '.[coverage]'
+
+    - name: Install JS Dependencies
+      if: ${{ steps.cache-node-modules.outputs.cache-hit != 'true' }}
+      run: jlpm --prefer-offline --frozen-lockfile
+
```

Is there a reason to install JS dependencies once more (already in l154)?

```suggestion
    - name: Install JS Dependencies
      if: ${{ steps.cache-node-modules.outputs.cache-hit != 'true' }}
      run: jlpm --prefer-offline --frozen-lockfile

```

---

yup!